test/node-rpc-test.js

```diff
@@ -178,6 +184,12 @@ describe('RPC', function() {
       assert.strictEqual(connectionsCnt, 1);
     });
 
+    it('should rpc getnettotals with peers', async () => {
+      const totals= await nclient.execute('getnettotals', []);
+      assert.notStrictEqual(totals.totalbytesrecv, 0);
+      assert.notStrictEqual(totals.totalbytessent, 0);
```

Comment:
Ok this is good but I think it can be better. A good, tight test will fail if anything is broken in the code we are covering. For example here I was curious what the non-zero value actually is. By adding some log messages to `peer.js` and running the test again I was able to determine all the packets exchanged with the peer and their sizes up to this point in the test (at this point its just the routine p2p handshake). Note that the total bytes sent/received isn't just `Packet.getSize()` (which is just the "payload") but also includes the command name itself, network magic bytes, and checksum. See [`framer.packet()`](https://github.com/bcoin-org/bcoin/blob/master/lib/net/framer.js#L38-L67)


| cmd | size |
|-|-|
| version | 123 |
| verack | 24 |
| sendcmpct | 33 |
| getaddr | 24 |
| addr | 55 |
| **TOTAL** | 259 |

Check my math on this, then try asserting that these values are indeed `259`. Is there anything that could make this test fail unpredictably? Can you make the test fail somehow by altering code in `pool.js` or `peer.js` ?

You should test your tests! The tests are here to protect us against bugs. If you intentionally add a bug somewhere, does the test fail?

Reply:
Yep, you are right. I did check the total bytes sent/received by logging and it is indeed 259. I have added the relevant assertion in the test file now and have squashed it into a single commit.